### PR TITLE
chore(manifest): simplify load balanced/backend service struct

### PR DESCRIPTION
**Pull request by iamhopaul123**
Simplify `BackendService` and `LoadbalancedWebService` by removing fields that are duplicated with the corresponding config struct. Addressing feedback https://github.com/aws/amazon-ecs-cli-v2/pull/970#pullrequestreview-419569244

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/manifest/backend_svc.go [hunk 1]
```diff
@@ -28,17 +28,15 @@ type BackendServiceProps struct {
...
-	Environments map[string]*backendServiceOverrideConfig `yaml:",flow"`
+	Service              `yaml:",inline"`
+	BackendServiceConfig `yaml:",inline"`
+	Environments         map[string]*BackendServiceConfig `yaml:",flow"`
 
 	parser template.Parser
 }
 
-type backendServiceOverrideConfig struct {
+// BackendServiceConfig holds the configuration for a backend service.
```
Comment (efekarakus) [inline internal/pkg/manifest/backend_svc.go:38]: Maybe: 
```suggestion
// BackendServiceConfig holds the configuration that can overriden per environments.
```

File: internal/pkg/manifest/backend_svc.go [hunk 2]
```diff
@@ -28,17 +28,15 @@ type BackendServiceProps struct {
...
 type BackendService struct {
-	Service      `yaml:",inline"`
-	Image        imageWithPortAndHealthcheck `yaml:",flow"`
-	TaskConfig   `yaml:",inline"`
-	LogConfig    `yaml:"logging,flow"`
-	Sidecar      `yaml:",inline"`
-	Environments map[string]*backendServiceOverrideConfig `yaml:",flow"`
+	Service              `yaml:",inline"`
+	BackendServiceConfig `yaml:",inline"`
+	Environments         map[string]*BackendServiceConfig `yaml:",flow"`
```
Comment (efekarakus) [inline internal/pkg/manifest/backend_svc.go:33]: Should we add a comment here explain why this one is a pointer, whereas the embedded struct isn't?

File: internal/pkg/manifest/backend_svc.go [hunk 3]
```diff
@@ -113,10 +111,7 @@ func (s BackendService) ApplyEnv(envName string) (*BackendService, error) {
 		return &s, nil
 	}
 	target := BackendService{
-		Image:      overrideConfig.Image,
-		TaskConfig: overrideConfig.TaskConfig,
-		LogConfig:  overrideConfig.LogConfig,
-		Sidecar:    overrideConfig.Sidecar,
+		BackendServiceConfig: *overrideConfig,
 	}
 	if err := mergo.Merge(&s, target, mergo.WithOverride, mergo.WithOverwriteWithEmptyValue); err != nil {
```
Comment (efekarakus) [inline internal/pkg/manifest/backend_svc.go:116]: I find our variable names a little bit confusing, shouldn't they be named the opposite lol where `s` is the real target, and `target` is the source. Maybe this?

```suggestion
        // Apply overrides to the original service s.
        err := mergo.Merge(&s, BackendService{
               BackendServiceConfig: *overrideConfig,
        }, mergo.WithOverride, mergo.WithOverwriteWithEmptyValue); 
        if err != nil {
```

File: internal/pkg/manifest/backend_svc.go (hunk 1 above)
Comment (iamhopaul123) [reply internal/pkg/manifest/backend_svc.go]: done!

File: internal/pkg/manifest/backend_svc.go (hunk 2 above)
Comment (iamhopaul123) [reply internal/pkg/manifest/backend_svc.go]: done!

File: internal/pkg/manifest/backend_svc.go (hunk 3 above)
Comment (iamhopaul123) [reply internal/pkg/manifest/backend_svc.go]: done!

**Comment by iamhopaul123** [comment]
@mergifyio refresh

**Pull request merged**